File: TravelBook/app/src/main/java/com/melihakkose/travelbook/DetailActivity.java
package com.melihakkose.travelbook;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;

//import static com.melihakkose.travelbook.MainActivity.selectedImage;

public class DetailActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        ImageView imageView=findViewById(R.id.imageView);
        TextView textViewPlaceName=findViewById(R.id.textViewPlaceName);
        TextView textViewProvinceName=findViewById(R.id.textViewProvinceName);

        Intent intent=getIntent();

        String place=intent.getStringExtra("placename");
        String province=intent.getStringExtra("provincename");
        textViewPlaceName.setText(place);
        textViewProvinceName.setText(province);
        Singleton singleton=Singleton.getInstance();
        imageView.setImageBitmap(singleton.getChosenImage());



        //imageView.setImageBitmap(selectedImage);

    }
}

File: TravelBook_Basic_Kotlin/app/src/main/java/com/melihakkose/travelbook_basic_kotlin/DetailActivity.kt
package com.melihakkose.travelbook_basic_kotlin

import android.graphics.BitmapFactory
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_detail.*

class DetailActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        val intent=intent
        val travelNames=intent.getStringExtra("name")
        textViewCity.text=travelNames

        //Global degisken ile tanimlanan bitmap i aktif etme kodu
        //imageViewCity.setImageBitmap(selectedGlobalBitmap)

        //Singleton
        /*
        val singleton=Singleton.Selected
        val selectedBitmap=singleton.selectedImage
        imageViewCity.setImageBitmap(selectedBitmap)

         */
        //Int degerleri intent ile alip bitmap ile decode ediyoruz.
        val travelImage=intent.getIntExtra("image",0)
        val selectedBitmap =BitmapFactory.decodeResource(applicationContext.resources,travelImage)
        imageViewCity.setImageBitmap(selectedBitmap)

    }
}